socket_test.go

```diff
@@ -1,77 +1,65 @@
+//go:build linux
 // +build linux
 
 package netlink
 
 import (
-	"log"
 	"net"
 	"os/user"
 	"strconv"
 	"syscall"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
 func TestSocketGet(t *testing.T) {
 	defer setUpNetlinkTestWithLoopback(t)()
 
 	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
-	if err != nil {
-		log.Fatal(err)
-	}
+	require.NoError(t, err)
```

Comment:
No need for this for this feature addition.
Existing testing code is readable just fine IMO.